pkg/balancer-js/src/pool-utils/swapper.ts

```diff
@@ -0,0 +1,149 @@
+import { Signer } from '@ethersproject/abstract-signer';
+import { BigNumber, BigNumberish } from '@ethersproject/bignumber';
+import { AddressZero, Zero } from '@ethersproject/constants';
+import { ContractTransaction, PayableOverrides } from '@ethersproject/contracts';
+import { Provider } from '@ethersproject/providers';
+import invariant from 'tiny-invariant';
+import { Vault, Vault__factory, WETH, WETH__factory } from '@balancer-labs/typechain';
+import { BatchSwapStep, FundManagement, SingleSwap, SwapKind } from '../types';
+
+/**
+ * Converts a batch swap which interacts with only a single pool to an equivalent simple swap
+ * @param swap - a SingleSwap object describing the pool to perform the swap against.
+ * @returns an equivalent batch swap
+ */
+export const convertSingleToBatchSwap = (
+  swap: SingleSwap,
+  limit: BigNumberish
+): { kind: SwapKind; swaps: BatchSwapStep[]; assets: string[]; limits: BigNumber[] } => {
+  const assets = [swap.assetIn, swap.assetOut];
```

Comment:
```suggestion
  const assets = [swap.assetIn, swap.assetOut];
```

These should be sorted